### ISSUE TITLE
some kind of problem with `:alpha:`

### ISSUE THREAD

**Issue by nikomatsakis** [nikomatsakis/lalrpop#244]
In chalk, we had a problem where the regex

```
r"([A-Za-z]|_)([A-Za-z0-9]|_)*"
```

did not match the string `Foo`. I'm not sure what's up with that. I have to narrow this down; it may be a problem in lalrpop, the regex-syntax crate, or the `RegexSet` code.

**Comment by nikomatsakis** [comment]
Here is the commit working around the problem:

https://github.com/nikomatsakis/chalk/pull/50/commits/8bc10387c710d19ed5091b5bae5f29a4e0afea01

**Comment by sgodwin-fbb** [comment]
I tried looking at this, but the regex you provided:

```
r"([A-Za-z]|_)([A-Za-z0-9]|_)*"
```

seems to have no problem matching `Foo`. I'm assuming you actually meant (as it shows in chalk)?

```
r"([:alpha:]|_)([:alnum:]|_)*"
```

If so, I may be wrong, but I think it should be:

```
r"([[:alpha:]]|_)([[:alnum:]]|_)*"
```

otherwise it is matching against any of the characters in `:alpha:` and `:alnum:`. Using that regex matches against `Foo` in a test grammar I made (whereas the previous one can actually match against `lm:::::`)

**Comment by sgodwincs** [comment]
I tried looking at this, but the regex you provided:

```
r"([A-Za-z]|_)([A-Za-z0-9]|_)*"
```

seems to have no problem matching `Foo`. I'm assuming you actually meant (as it shows in chalk)?

```
r"([:alpha:]|_)([:alnum:]|_)*"
```

If so, I may be wrong, but I think it should be:

```
r"([[:alpha:]]|_)([[:alnum:]]|_)*"
```

otherwise it is matching against any of the characters in `:alpha:` and `:alnum:`. Using that regex matches against `Foo` in a test grammar I made (whereas the previous one can actually match against `lm:::::`)
